### PR TITLE
integration tests for get_repo_info

### PR DESCRIPTION
**Pull request by mandylan75**
#46

### CODE REVIEW

File: tests/test_get_repo_info_integration.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+        description=None,
+        homepage=None,
+        readme_content="Just a plain readme with nothing special.",
+        files={},
+        citation_yaml=None,
+        languages={},
+        forks_count=0,
+        private=True,
+        fork=True,
+        archived=True,
```
Comment (egrace479) [inline tests/test_get_repo_info_integration.py]: It's a bit odd testing these three together on the bare-repo, since overall it's an odd combination maybe keep private true and test the others in one more test with the more full example repo.

**Review by egrace479 — COMMENTED** [review]
One suggestion to make an intermediate repo test.

File: tests/test_get_repo_info_integration.py [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+    )
+    repo.get_license.side_effect = GithubException(404, "Not Found", None)
+
+    result = exporter.get_repo_info(repo, existing_df=None)
+
+    assert result["Visibility"] == "Public"
+    assert result["Is Fork"] == "Yes"
+    assert result["Archived"] == "Yes"
```
Comment (egrace479) [inline tests/test_get_repo_info_integration.py]:
```suggestion
    assert result["Archived"] == "Yes"
    assert result[".gitignore"] == "Yes"
    assert result["Package Requirements"] == "Yes"
```
And I think description would return N/A?

File: tests/test_get_repo_info_integration.py [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    assert result[".gitignore"] == "No"
+    assert result["Package Requirements"] == "No"
+    assert result["CITATION"] == "No"
+    assert result["Language"] == "N/A"
+    assert result["Visibility"] == "Private"
+    assert result["Has Forks"] == "No"
+    assert result["Website Reference"] == "No"
+    assert result["Dataset"] == "No"
+    assert result["Model"] == "No"
+    assert result["Paper Association"] == "No"
```
Comment (egrace479) [inline tests/test_get_repo_info_integration.py:241]:
```suggestion
    assert result["Paper Association"] == "No"
    assert result[".zenodo.json"] == "No"
    assert result["CONTRIBUTING"] == "No"
    assert result["AGENTS"] == "No"
```

**Review by egrace479 — COMMENTED** [review]
It might be easier to have a list of expected values that should be returned or order them consistently across tests, since we want to ensure all provide the expected return value for each test.

File: tests/test_get_repo_info_integration.py (hunk 2 above)
Comment (mandylan75) [reply tests/test_get_repo_info_integration.py]: The mock defaults to `description="A cool research project"` so `get_repo_info` returns it as it is since it only falls back to "N/A" if the description is None or empty. I added `assert result["Description"] == "A cool research project"` to test 3 to make that explicit so theres no changes needed for the N/A behavior since that's already covered in test 2 with `description=None`.

File: tests/test_get_repo_info_integration.py [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+doesn't silently change the exported data.
+"""
+
+from datetime import datetime, timezone
+from unittest.mock import MagicMock
+
+from github import GithubException
+
+import gh_repo_exporter as exporter
+
```
[the remark this answers is not in the document]
Comment (mandylan75) [reply tests/test_get_repo_info_integration.py]: Should I implement this?

[the remark this answers is not in the document]
Comment (egrace479) [reply tests/test_get_repo_info_integration.py]: Or we could just use `datetime.now(timezone.utc)` or `datetime.now(timezone.utc)-timedelta(days=3)` or something like that for the `updated_at` setting. We should then use `updated_at=datetime(2024, 1, 1, tzinfo=timezone.utc)` as a mock on one of the tests to get an "inactive" return.

File: tests/test_get_repo_info_integration.py [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+    assert result["README"] == "Yes"
+    assert result["License"] == "No"
+    assert result[".gitignore"] == "No"
+    assert result["Package Requirements"] == "No"
+    assert result["CITATION"] == "No"
+    assert result[".zenodo.json"] == "No"
+    assert result["CONTRIBUTING"] == "No"
+    assert result["AGENTS"] == "No"
+    assert result["Language"] == "N/A"
+    assert result["Visibility"] == "Private"
```
Comment (egrace479) [inline tests/test_get_repo_info_integration.py:233]:
```suggestion
    assert result["Visibility"] == "Private"
    assert result["Is Fork"] == "No"
```

File: tests/test_get_repo_info_integration.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (egrace479) [reply tests/test_get_repo_info_integration.py]: The inactive key should then be added as appropriate for the last two tests.

**Review by egrace479 — CHANGES_REQUESTED** [review]
Repo name, created (date + creator), last updated, stars, and No. of branches need to be added to the checks for the last two tests also.

**Comment by egrace479** [comment]
Can you also rename the file to `test_get_gh_repo_info_integration.py`? We'll want to duplicate it for a `test_get_hf_repo_info_integration.py` with some tweaks, since there are shared columns, some of which do use the same functions.

File: tests/test_get_hf_repo_info_integration.py [hunk 6]
```diff
@@ -0,0 +1,226 @@
+"""
+Integration test for hf_repo_exporter.get_repo_info().
+
+Builds mocked HuggingFace repo and API objects (no network calls) and checks
+get_repo_info() against a frozen "golden" expected dict, so that
+refactoring (splitting modules, moving into src/repo_exporter, etc.)
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
Builds mocked Hugging Face repo and API objects (no network calls) and checks
get_repo_info() against a frozen "golden" expected dict, so that
refactoring (splitting modules, moving into src/repo-exporter, etc.)
```

File: tests/test_get_hf_repo_info_integration.py [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+Homepage: https://example.org/cool-dataset
+Repository: https://github.com/Imageomics/cool-dataset
+Paper: https://arxiv.org/abs/1234.5678
+"""
+
+
+def test_get_repo_info_matches_expected_output():
+    """Golden test: a fully-populated dataset repo should produce this exact row."""
+    repo = make_mock_repo(
+        tags=["dataset:imageomics/cool-data", "doi:10.5281/zenodo.1234567"],
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        tags=["dataset:imageomics/cool-data", "doi:10.57967/hf/1234567"],
```
DOIs here are provided through HF

File: tests/test_get_hf_repo_info_integration.py [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+        "License": "mit",
+        "Visibility": "Public",
+        "Inactive": "No",
+        "Homepage": '=HYPERLINK("https://example.org/cool-dataset", "https://example.org/cool-dataset")',
+        "Repo": '=HYPERLINK("https://github.com/Imageomics/cool-dataset", "https://github.com/Imageomics/cool-dataset")',
+        "Paper": '=HYPERLINK("https://arxiv.org/abs/1234.5678", "https://arxiv.org/abs/1234.5678")',
+        "Associated Datasets": "imageomics/cool-data",
+        "Associated Models": "No",
+        "Associated Spaces": "imageomics/cool-space",
+        "DOI": "10.5281/zenodo.1234567",
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        "DOI": "10.57967/hf/1234567",
```

File: tests/test_get_hf_repo_info_integration.py [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+
+    with patch("hf_repo_exporter.hf_hub_download", return_value="/fake/README.md"), \
+         patch("builtins.open", mock_open(read_data="Just a plain readme with nothing special.")):
+        result = exporter.get_repo_info(api, repo, "model")
+
+    assert result["Repository Name"] == '=HYPERLINK("https://huggingface.co/imageomics/bare-repo", "imageomics/bare-repo")'
+    assert result["Repository Type"] == "model"
+    assert result["Description"] == "N/A"
+    assert result["Date Created"] == "2022-01-01"
+    assert result["Last Updated"] == "2025-12-01"
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py:186]: This will become inactive in December. Please implement the fixed time [suggestion from Copilot](https://github.com/Imageomics/repo-exporter/pull/53#discussion_r3406009947) for this file as well.

File: tests/test_get_repo_info_integration.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (egrace479) [reply tests/test_get_repo_info_integration.py]: Yes, please implement the suggestion

**Review by egrace479 — COMMENTED** [review]
A few notes and suggestions.

File: tests/test_get_hf_repo_info_integration.py (hunk 7 above)
Comment (mandylan75) [reply tests/test_get_hf_repo_info_integration.py]: Thank you for letting me know! Fixing that right now.

File: tests/test_get_gh_repo_info_integration.py [hunk 10]
```diff
@@ -0,0 +1,292 @@
...
+        citation_yaml=FULL_CITATION,
+    )
+
+    result = exporter.get_repo_info(repo, existing_df=None)
+
+    expected = {
+        "Repository Name": '=HYPERLINK("https://github.com/Imageomics/cool-project", "cool-project")',
+        "Description": "A cool research project",
+        "Date Created": "2022-01-01",
+        "Last Updated": repo.updated_at.strftime("%Y-%m-%d"),
```
Comment (egrace479) [inline tests/test_get_gh_repo_info_integration.py]:
```suggestion
        "Last Updated": "2026-01-01",
```

File: tests/test_get_gh_repo_info_integration.py [hunk 11]
```diff
@@ -0,0 +1,292 @@
...
+    languages=None,
+    readme_content="",
+    files=None,
+    citation_yaml=None,
+    creator=("Jane Doe", "janedoe"),
+):
+    """Build a MagicMock that mimics a PyGithub Repository object."""
+    if updated_at is None:
+        updated_at = datetime.now(timezone.utc) - timedelta(days=3)
+
```
Comment (egrace479) [inline tests/test_get_gh_repo_info_integration.py]:
```suggestion
```

File: tests/test_get_gh_repo_info_integration.py [hunk 12]
```diff
@@ -0,0 +1,292 @@
...
+        self.author = FakeAuthor(name, login)
+        self.weeks = [FakeWeek(additions, deletions)]
+
+
+def make_mock_repo(
+    *,
+    name="cool-project",
+    description="A cool research project",
+    created_at=datetime(2022, 1, 1, tzinfo=timezone.utc),
+    updated_at=None,
```
Comment (egrace479) [inline tests/test_get_gh_repo_info_integration.py]:
```suggestion
    updated_at=datetime(2026, 1, 1, tzinfo=timezone.utc),
```

File: tests/test_get_hf_repo_info_integration.py [hunk 13]
```diff
@@ -0,0 +1,239 @@
...
+        self.is_pull_request = is_pull_request
+        self.status = status
+
+
+def make_mock_repo(
+    *,
+    repo_id="imageomics/cool-dataset",
+    description="A cool research dataset",
+    created_at=datetime(2022, 1, 1, tzinfo=timezone.utc),
+    last_modified=None,
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
    last_modified=datetime(2026, 1, 1, tzinfo=timezone.utc),
```

File: tests/test_get_hf_repo_info_integration.py [hunk 14]
```diff
@@ -0,0 +1,239 @@
...
+    with patch("hf_repo_exporter.hf_hub_download", return_value="/fake/README.md"), \
+         patch("builtins.open", mock_open(read_data=FULL_README)):
+        result = exporter.get_repo_info(api, repo, "dataset")
+
+    expected = {
+        "Repository Name": '=HYPERLINK("https://huggingface.co/datasets/imageomics/cool-dataset", "datasets/imageomics/cool-dataset")',
+        "Repository Type": "dataset",
+        "Description": "A cool research dataset",
+        "Date Created": "2022-01-01",
+        "Last Updated": repo.lastModified.strftime("%Y-%m-%d"),
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        "Last Updated": "2026-01-01",
```

**Review by egrace479 — COMMENTED** [review]
I think I got all the locations we were discussing here.

File: tests/test_get_hf_repo_info_integration.py [hunk 15]
```diff
@@ -0,0 +1,239 @@
...
+Homepage: https://example.org/cool-dataset
+Repository: https://github.com/Imageomics/cool-dataset
+Paper: https://arxiv.org/abs/1234.5678
+"""
+
+
+def test_get_repo_info_matches_expected_output():
+    """Golden test: a fully-populated dataset repo should produce this exact row."""
+    repo = make_mock_repo(
+        tags=["dataset:imageomics/cool-data", "doi:10.57967/hf.1234567"],
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        tags=["dataset:imageomics/cool-data-source", "doi:10.57967/hf.1234567"],
```

File: tests/test_get_hf_repo_info_integration.py [hunk 16]
```diff
@@ -0,0 +1,239 @@
...
+        "Likes": 42,
+        "# of Open PRs": 2,
+        "README": "Yes",
+        "License": "mit",
+        "Visibility": "Public",
+        "Inactive": "No",
+        "Homepage": '=HYPERLINK("https://example.org/cool-dataset", "https://example.org/cool-dataset")',
+        "Repo": '=HYPERLINK("https://github.com/Imageomics/cool-dataset", "https://github.com/Imageomics/cool-dataset")',
+        "Paper": '=HYPERLINK("https://arxiv.org/abs/1234.5678", "https://arxiv.org/abs/1234.5678")',
+        "Associated Datasets": "imageomics/cool-data",
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        "Associated Datasets": "imageomics/cool-data-source",
```
It shouldn't return itself; works with above suggestion.

File: tests/test_get_hf_repo_info_integration.py [hunk 17]
```diff
@@ -0,0 +1,239 @@
...
+        "Associated Models": "No",
+        "Associated Spaces": "imageomics/cool-space",
+        "DOI": "10.57967/hf.1234567",
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        "DOI": "10.57967/hf/1234567",
```

File: tests/test_get_hf_repo_info_integration.py [hunk 18]
```diff
@@ -0,0 +1,239 @@
...
+    private=False,
+    likes=42,
+    card_data=None,
+    tags=None,
+    doi_attr=None,
+):
+    """Build a MagicMock that mimics a huggingface_hub repo info object."""
+    if last_modified is None:
+        last_modified = datetime.now(timezone.utc) - timedelta(days=3)
+
```
Comment (mandylan75) [inline tests/test_get_hf_repo_info_integration.py]: Thank you!

File: tests/test_get_hf_repo_info_integration.py [hunk 19]
```diff
@@ -0,0 +1,239 @@
...
+    with patch("hf_repo_exporter.hf_hub_download", return_value="/fake/README.md"), \
+         patch("builtins.open", mock_open(read_data=FULL_README)):
+        result = exporter.get_repo_info(api, repo, "space")
+
+    assert result["Repository Name"] == '=HYPERLINK("https://huggingface.co/spaces/imageomics/cool-space", "spaces/imageomics/cool-space")'
+    assert result["Repository Type"] == "space"
+    assert result["Description"] == "A cool research dataset"
+    assert result["Date Created"] == "2022-01-01"
+    assert result["Last Updated"] == "2020-01-01"
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
    assert result["Date Created"] == "2022-01-01"
    assert result["Last Updated"] == "2024-01-01"
```
This is created after it was last updated.

File: tests/test_get_hf_repo_info_integration.py [hunk 20]
```diff
@@ -0,0 +1,239 @@
...
+    assert result["DOI"] == "No"
+
+
+def test_get_repo_info_space_type_and_inactive_repo():
+    """Intermediate case: a Space repo type that is also inactive (last modified
+    over a year ago). Keeps repo_type URL/display logic and Inactive isolated
+    from the bare/minimal repo's edge cases."""
+    repo = make_mock_repo(
+        repo_id="imageomics/cool-space",
+        last_modified=datetime(2020, 1, 1, tzinfo=timezone.utc),
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        last_modified=datetime(2024, 1, 1, tzinfo=timezone.utc),
```
to go with timeline fix

File: tests/test_get_hf_repo_info_integration.py [hunk 21]
```diff
@@ -0,0 +1,239 @@
...
+    assert result["Associated Spaces"] == "No"
+    assert result["DOI"] == "No"
+
+
+def test_get_repo_info_space_type_and_inactive_repo():
+    """Intermediate case: a Space repo type that is also inactive (last modified
+    over a year ago). Keeps repo_type URL/display logic and Inactive isolated
+    from the bare/minimal repo's edge cases."""
+    repo = make_mock_repo(
+        repo_id="imageomics/cool-space",
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py:209]:
```suggestion
        repo_id="imageomics/cool-space",
        description="A cool research demo",
```

File: tests/test_get_hf_repo_info_integration.py [hunk 22]
```diff
@@ -0,0 +1,239 @@
...
+    )
+    api = make_mock_api(open_pr_count=0)
+
+    with patch("hf_repo_exporter.hf_hub_download", return_value="/fake/README.md"), \
+         patch("builtins.open", mock_open(read_data=FULL_README)):
+        result = exporter.get_repo_info(api, repo, "space")
+
+    assert result["Repository Name"] == '=HYPERLINK("https://huggingface.co/spaces/imageomics/cool-space", "spaces/imageomics/cool-space")'
+    assert result["Repository Type"] == "space"
+    assert result["Description"] == "A cool research dataset"
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
    assert result["Description"] == "A cool research demo"
```
Matches suggestion above (for consistency of type).

**Review by egrace479 — COMMENTED** [review]
A few more suggestions for consistency

File: tests/test_get_hf_repo_info_integration.py [hunk 23]
```diff
@@ -0,0 +1,237 @@
...
+Homepage: https://example.org/cool-dataset
+Repository: https://github.com/Imageomics/cool-dataset
+Paper: https://arxiv.org/abs/1234.5678
+"""
+
+
+def test_get_repo_info_matches_expected_output():
+    """Golden test: a fully-populated dataset repo should produce this exact row."""
+    repo = make_mock_repo(
+        tags=["dataset:imageomics/cool-data-source", "doi:10.57967/hf.1234567"],
```
Comment (egrace479) [inline tests/test_get_hf_repo_info_integration.py]:
```suggestion
        tags=["dataset:imageomics/cool-data-source", "doi:10.57967/hf/1234567"],
```
I missed this with the expected output change.

Comment (mandylan75) [reply tests/test_get_hf_repo_info_integration.py]: Thank you for the help!

**Review by egrace479 — APPROVED** [review]
